pkgs/io_file/lib/src/file_system.dart

```diff
@@ -223,6 +223,18 @@ abstract class FileSystem {
   String get currentDirectory;
   set currentDirectory(String path);
 
+  /// Checks if a file system object exists for the given path.
+  ///
+  /// Returns `true` if a file, directory, or link exists at the given path,
+  /// and `false` otherwise.
+  ///
+  /// If `path` is a symbolic link, `exists` returns `false` if the link is
+  /// broken (i.e. the target of the link does not exist).
+  ///
+  /// On Windows, calling `exists` on a named pipe may cause the server to close
```

Comment:
This sounds unexpected: it is surprising that `exists` can cause change to any state.

Reply:
`exists` call `CreateFile`, which opens the file. `dart:io` (sometimes) has the same behavior (and so does Python, Rust, etc.)

Reply:
I was thinking about making `exists` not follow the symlink so that it doesn't need to open the file. But that does against the current behavior of `dart:io` (and Python and Rust...)